desc/magnetic_fields/_core.py

```diff
@@ -50,28 +51,32 @@ def biot_savart_general(re, rs, J, dV):
         current density vector at source points, in cartesian.
     dV : ndarray, shape(n_src_pts)
         volume element at source points
+    chunk_size : int or None
+        Size to split computation into chunks.
+        If no chunking should be done or the chunk size is the full input
+        then supply ``None``.
 
     Returns
     -------
-    B : ndarray, shape(n,3)
+    B : ndarray, shape(n_eval_pts, 3)
         magnetic field in cartesian components at specified points
+
     """
     re, rs, J, dV = map(jnp.asarray, (re, rs, J, dV))
     assert J.shape == rs.shape
-    JdV = J * dV[:, None]
-    B = jnp.zeros_like(re)
-
-    def body(i, B):
-        r = re - rs[i, :]
-        num = jnp.cross(JdV[i, :], r, axis=-1)
-        den = jnp.linalg.norm(r, axis=-1) ** 3
-        B = B + jnp.where(den[:, None] == 0, 0, num / den[:, None])
-        return B
+    JdV = J * dV[:, jnp.newaxis]
+
+    def biot(re):
+        dr = rs - re[..., jnp.newaxis, :]
+        num = jnp.cross(dr, JdV, axis=-1)
+        den = jnp.linalg.norm(dr, axis=-1, keepdims=True) ** 3
+        return safediv(num, den).sum(axis=-2) * mu_0 / (4 * jnp.pi)
 
-    return scipy.constants.mu_0 / 4 / jnp.pi * fori_loop(0, J.shape[0], body, B)
+    # It is more efficient to sum over the sources in batches of evaluation points.
+    return batch_map(biot, re, chunk_size)
 
 
-def biot_savart_general_vector_potential(re, rs, J, dV):
+def biot_savart_general_vector_potential(re, rs, J, dV, chunk_size=30):
```

Comment:
This function is almost identical to `biot_savart_general` except for 2 lines -- could we combine them with an input flag to distinguish when to use one method or the other? That would give less redundant code.

Reply:
I'd prefer that cosmetic change to be a different PR, unless you wanted to do it. There's about 50 different places you would have to make that change.

Reply:
yea I don't think we need to make this change right now.